### PR TITLE
added ability to join rooms

### PR DESCRIPTION
**Pull request by bumblebot1**
Players can now join existing rooms. After joining a room they are not redirected yet so you can only see the change if you have another session(like incognito) open with another user.(The redirect is part of a different story)

### CODE REVIEW

**Comment by JamieZafar** [comment]
:new_moon_with_face:

File: src/server/cloak-server.spec.js [hunk 1]
```diff
@@ -100,4 +101,17 @@ describe('cloak server', function() {
         cloakConfig.messages.createRoom('TEST_ROOM_NAME', user);
         expect(cloak.createRoom).toHaveBeenCalledWith('TEST_ROOM_NAME');
     });
+
+    it('joinRoom: finds the room that has the ID passed as argument', function() {
```
Comment (kenneth-gray) [inline src/server/cloak-server.spec.js]: The better way to do this is nest a new describe block:
```
describe('joinRoom', () => {
    it('finds the room that has the ID passed as argument', () => {});
});
```

File: src/server/cloak-server.spec.js [hunk 2]
```diff
@@ -100,4 +101,17 @@ describe('cloak server', function() {
...
+        require('./cloak-server')({});
```
Comment (kenneth-gray) [inline src/server/cloak-server.spec.js]: This is called as the first thing in every test as far as I can tell, so it should probably be pulled out into either an existing beforeEach or a new one.

File: src/server/cloak-server.js [hunk 3]
```diff
@@ -37,6 +37,11 @@ module.exports = function(expressServer) {
             createRoom: (arg, user) => {
                 cloak.createRoom(arg);
                 fireRoomListReload();
+            },
+            joinRoom: (arg, user) => {
```
Comment (kenneth-gray) [inline src/server/cloak-server.js]: I think that for better readability of this file these functions should be defined separately and assigned to the properties:
```
...
joinRoom: joinRoom
...

function joinRoom() {
    ...
}
```

**Comment by JamieZafar** [comment]
:ship: dat :hankey: ?

**Comment by JamieZafar** [comment]
:pray: :shipit:

**Pull request merged**